repositories.py

```diff
@@ -51,9 +51,10 @@ def update_teams(self):
             utils.remove_response_from_cache(uri)
             self._teams = []
             return
-        if r.status_code == 200:
-            utils.cache_response(r)
-            self._teams = list(map(lambda r: r['name'], json.loads(r.text)))
+        if r.status_code == 304:
+            r = utils.cached_response(uri)
+        utils.cache_response(r)
```

Comment:
Check for 200 is still needed here. API can return 404 or 500 or whatever and it will be cached